### ISSUE TITLE
Update paths of moved items

### ISSUE THREAD

**Issue by techalchemy** [sarugaku/pip-shims#35]
The following items have moved:
- `SafeFileCache` -> `network.cache`
- `Link` -> `models.link.Link`
- `path_to_url` -> `utils.url`
- `url_to_path` -> `utils.url`
- `SourceDistribution` -> `distributions.source.legacy`

Other notable changes:
- For some reason the resolver no longer accepts `isolated`, `wheel_cache` as arguments and can no longer generate install requirements. Instead you must pass in a partial function yourself that can generate install requirements:

```python
index_urls = [pip_options.index_url] + pip_options.extra_index_urls
search_scope = SearchScope.create(
    find_links=pip_options.find_links, index_urls=index_urls
)
link_collector = LinkCollector(session=session, search_scope=search_scope)
selection_prefs = SelectionPreferences(
    True,
    allow_all_prereleases=False,
    format_control=None,
    prefer_binary=False,
    ignore_requires_python=False,
)
target_python = TargetPython()
candidate_prefs = CandidatePreferences(
    prefer_binary=selection_prefs.prefer_binary,
    allow_all_prereleases=selection_prefs.allow_all_prereleases,
)
finder_args = {
    "candidate_prefs": candidate_prefs,
    "link_collector": link_collector,
    "target_python": target_python,
    "allow_yanked": selection_prefs.allow_yanked,
    "format_control": selection_prefs.format_control,
    "ignore_requires_python": selection_prefs.ignore_requires_python,
}
finder = PackageFinder(**finder_args)
build_dir = tmpdir.mkdir("build_dir")
source_dir = tmpdir.mkdir("source_dir")
download_dir = tmpdir.mkdir("download_dir")
wheel_download_dir = CACHE_DIR.mkdir("wheels")
wheel_cache = WheelCache(USER_CACHE_DIR, FormatControl(None, None))
kwargs = {
    "build_dir": build_dir.strpath,
    "src_dir": source_dir.strpath,
    "download_dir": download_dir.strpath,
    "wheel_download_dir": wheel_download_dir.strpath,
    "wheel_cache": wheel_cache,
}
ireq = InstallRequirement.from_editable(
    "git+https://github.com/urllib3/urllib3@1.23#egg=urllib3"
)
ireq.populate_link(finder, False, False)
ireq.ensure_has_source_dir(kwargs["src_dir"])
make_install_req = partial(
    install_req_from_req_string,
    isolated=False,
    wheel_cache=wheel_cache,
    # use_pep517=use_pep517,
)
resolver_kwargs = {
    "finder": finder,
    "session": session,
    "upgrade_strategy": "to-satisfy-only",
    "make_install_req": make_install_req,
    "force_reinstall": False,
    "ignore_dependencies": False,
    "ignore_requires_python": False,
    "ignore_installed": True,
    "use_user_site": False,
}
```

- The Wheel Builder no longer accepts a finder
- `unpack_url` no longer accepts `only_download` as an argument

And probably most kind of strangely there are now about 5 objects you need build and pass into a `PackageFinder` rather than simply passing in arguments:
- `SearchScope`: seems to just specify `find_links` and `index_urls`, is its own object in its own file
- `LinkCollector`: accepts a `session` and a `SearchScope`, not too sure since I thought it just accumulated links but I'm guessing now that it actually queries for them
- `SelectionPreferences`: Accepts `allow_all_prereleases`, `format_control`, `prefer_binary`, `ignore_requires_python`
- `TargetPython`: Accepts a bunch of arguments describing the valid tags to include
- `CandidatePreferences`: Takes `allow_all_prereleases` and `prefer_binary` from `SelectionPreferences`
- `PackageFinder`: Takes `candidate_prefs`, `link_collector`, `target_python`, `allow_yanked`, `format_control`, and `ignore_requires_python` -- these last build `SelectionPreferences` so I dion't know why they made an exception here rather than demanding the whole object be built first...

Finders no longer carry a `session` instance either, which we used to use for persistence

Basically everything in pip is super unstable atm and currently trending toward lots of 'providers' and i'm suspecting we will see some factories soon :)

**Comment by pradyunsg** [comment]
> `SourceDistribution` -> `distributions.source.legacy`

I moved this back for 20.0 btw. :P
